pkg/sql/opt/bench: BenchmarkExecBuild failed

pkg/sql/opt/bench.BenchmarkExecBuild [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/20469786?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/20469786?buildTab=artifacts#./artifacts/microbench/experiment) on master @ [298633de384ce4aa4cf65d7d964422459a7c88a6](https://github.com/cockroachdb/cockroach/commits/298633de384ce4aa4cf65d7d964422459a7c88a6):


```
goos: linux
goarch: amd64
cpu: Intel(R) Xeon(R) CPU @ 2.60GHz
BenchmarkExecBuild
BenchmarkExecBuild/kv-read
BenchmarkExecBuild/kv-read-32         	 1400631	       852.6 ns/op	     744 B/op	       7 allocs/op
BenchmarkExecBuild/kv-read-const
BenchmarkExecBuild/kv-read-const-32   	 1399408	       854.8 ns/op	     744 B/op	       7 allocs/op
BenchmarkExecBuild/sysbench-update-index
BenchmarkExecBuild/sysbench-update-index-32         	  549591	      2154 ns/op	    1336 B/op	      13 allocs/op
BenchmarkExecBuild/sysbench-update-non-index
BenchmarkExecBuild/sysbench-update-non-index-32     	  646778	      1824 ns/op	    1272 B/op	      12 allocs/op
BenchmarkExecBuild/tpcc-new-order
BenchmarkExecBuild/tpcc-new-order-32                	  899154	      1290 ns/op	     928 B/op	       8 allocs/op
BenchmarkExecBuild/tpcc-delivery
BenchmarkExecBuild/tpcc-delivery-32                 	 1000000	      1108 ns/op	     728 B/op	       8 allocs/op
BenchmarkExecBuild/tpcc-stock-level
BenchmarkExecBuild/tpcc-stock-level-32              	  356102	      2975 ns/op	    1512 B/op	      17 allocs/op
BenchmarkExecBuild/many-columns-and-indexes-a
BenchmarkExecBuild/many-columns-and-indexes-a-32    	  667036	      1587 ns/op	    1128 B/op	      11 allocs/op
BenchmarkExecBuild/many-columns-and-indexes-b
BenchmarkExecBuild/many-columns-and-indexes-b-32    	  463657	      2237 ns/op	    1416 B/op	      15 allocs/op
BenchmarkExecBuild/many-columns-and-indexes-c
BenchmarkExecBuild/many-columns-and-indexes-c-32    	   47368	     23497 ns/op	   22656 B/op	      80 allocs/op
BenchmarkExecBuild/many-columns-and-indexes-d
BenchmarkExecBuild/many-columns-and-indexes-d-32    	   56929	     20292 ns/op	   10816 B/op	      32 allocs/op
BenchmarkExecBuild/many-columns-and-indexes-e
BenchmarkExecBuild/many-columns-and-indexes-e-32    	  554088	      1878 ns/op	    2080 B/op	      10 allocs/op
    bench_test.go:890: no Annotations provided
--- FAIL: BenchmarkExecBuild
FAIL

I250915 18:37:15.135572 1 (gostd) rand.go:268  [-] 1  random seed: 3310659265101134434
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*BenchmarkExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54479

--- a/pkg/sql/opt/bench/bench_test.go
+++ b/pkg/sql/opt/bench/bench_test.go
@@ -854,6 +854,7 @@ func newHarness(tb testing.TB, query benchQuery, schemas []string) *harness {
 		semaCtx: tree.MakeSemaContext(nil /* resolver */),
 		evalCtx: eval.MakeTestingEvalContext(cluster.MakeTestingClusterSettings()),
 	}
+	h.evalCtx.Annotations = &h.semaCtx.Annotations
 
 	// Set session settings to their global defaults.
 	if err := sql.TestingResetSessionVariables(h.ctx, h.evalCtx); err != nil {

